homeassistant/components/sensor/teksavvy.py

```diff
@@ -143,17 +145,25 @@ def async_update(self):
         if req.status != 200:
             _LOGGER.error("Request failed with status: %u", req.status)
             return False
-        data = yield from req.json()
-        for (api, ha_name) in API_HA_MAP:
-            self.data[ha_name] = float(data["value"][0][api])
-        on_peak_download = self.data["onpeak_download"]
-        on_peak_upload = self.data["onpeak_upload"]
-        off_peak_download = self.data["offpeak_download"]
-        off_peak_upload = self.data["offpeak_upload"]
-        limit = self.data["limit"]
-        self.data["usage"] = 100*on_peak_download/self.bandwidth_cap
-        self.data["usage_gb"] = on_peak_download
-        self.data["onpeak_total"] = on_peak_download + on_peak_upload
-        self.data["offpeak_total"] = off_peak_download + off_peak_upload
-        self.data["onpeak_remaining"] = limit - on_peak_download
-        return True
+        else:
+            try:
+                data = yield from req.json()
+                for (api, ha_name) in API_HA_MAP:
+                    self.data[ha_name] = float(data["value"][0][api])
+                on_peak_download = self.data["onpeak_download"]
+                on_peak_upload = self.data["onpeak_upload"]
+                off_peak_download = self.data["offpeak_download"]
+                off_peak_upload = self.data["offpeak_upload"]
+                limit = self.data["limit"]
+                # Support "unlimited" users
+                self.data["usage"] = 100*on_peak_download/self.bandwidth_cap\
+                    if self.bandwidth_cap > 0 else 0
+                self.data["usage_gb"] = on_peak_download
+                self.data["onpeak_total"] = on_peak_download + on_peak_upload
+                self.data["offpeak_total"] = off_peak_download +\
+                    off_peak_upload
+                self.data["onpeak_remaining"] = limit - on_peak_download
+                return True
+            except ValueError:
```

Comment:
What can cause a JSON decode error?

Reply:
It's an edge case, assuming the API responds with HTTP OK/200 but the payload data isn't proper JSON. Then this would catch it.